pkg/auth/oauth2authenticator_test.go

```diff
@@ -205,3 +206,62 @@ func Test_AddAuthenticationHeader_expiredToken_somebodyUpdated(t *testing.T) {
 	assert.Equal(t, *newToken, *actualToken)
 	assert.Equal(t, *newToken, *authenticator.(*oAuth2Authenticator).token)
 }
+
+func Test_determineGrantType(t *testing.T) {
+	config := configuration.NewInMemory()
+	expected := AuthorizationCodeGrant
+	actual := determineGrantType(config)
+	assert.Equal(t, expected, actual)
+
+	config.Set(PARAMETER_CLIENT_SECRET, "secret")
+	expected = AuthorizationCodeGrant
+	actual = determineGrantType(config)
+	assert.Equal(t, expected, actual)
+
+	config.Set(PARAMETER_CLIENT_ID, "id")
+	expected = ClientCredentialsGrant
+	actual = determineGrantType(config)
+	assert.Equal(t, expected, actual)
+}
```

Comment:
As far as I understand, this test is communicating that the order of the assertions are linked given that for the assertion in line **224** to be true, line **216** must have happened, right? 

Is it correct to say that this test is asserting the case when one of the required params for grant type to be `ClientCredentialsGrant` is missing, the test should fail?

Would it be worth creating units that stress:
- **neither** *client ID* nor *client secret* is set and the expectation is `AuthorizationCodeGrant`.
- **only** the *client secret* is set and the expectation is `AuthorizationCodeGrant`.
- **only** the *client ID* is set and the expectation is `AuthorizationCodeGrant`
- **both** are set and the and the expectation is `ClientCredentialsGrant`

via parametrise test like https://github.com/snyk/issue-policies/blob/main/internal/core/enums/class_type_test.go?